test_suite/tests/ui/fail_non_static_lifetime.rs

```diff
@@ -0,0 +1,12 @@
+use scale_info::TypeInfo;
+
+#[derive(TypeInfo)]
+struct Me<'a> {
+    me: &'a Me<'a>,
+}
+
+fn assert_type_info<T: TypeInfo + 'static>() {}
+
+fn main() {
+    assert_type_info::<Me>();
```

Comment:
What we really want is that `assert_type_info::<Me<'static>()` works, with an impl looking like:

```
impl TypeInfo for Me<'static> {
    type Identity = Self;
    fn type_info() -> scale_info::Type {
        // todo: .field_of<&'static Me<'static>>,
        unimplemented!()
    }
}
```

However I realise this is out of the scope of this PR. I did something like this in my closed generics PR: https://github.com/paritytech/scale-info/pull/12/files#diff-60da9bde79deaa50c0ce6f42d1f47205ba31f448dc2713cc746339343581b699R113

Reply:
Opened https://github.com/paritytech/scale-info/issues/38